portal/src/main/webapp/js/src/pancancer_study_summary/pancancer_study_summary.js

```diff
@@ -790,12 +790,20 @@ function DataManagerPresenter(dmInitCallBack)
 			
 			//do the next call:
 			console.log(new Date() + ": CALL to get sample clinical atttributes (cancer types)");
-			return window.QuerySession.getSampleClinicalData(["CANCER_TYPE","CANCER_TYPE_DETAILED"]);
+			//this and next call below are in parallel, so the last one should call the dmInitCallBack:
+			nrResponsesReceived++;
+			if (nrResponsesReceived == 2) {
+				dmInitCallBack(self);
+			}
```

Comment:
instead, you could do:

At line 769:
```
var genomic_data_fetched = new $.Deferred();
var clinical_data_fetched = new $.Deferred();
```

then here, ```genomic_data_fetched.resolve()```. Below, ```clinical_data_fetched.resolve()```. And instead of calling the callback,

``` $.when(genomic_data_fetched, clinical_data_fetched).then(function() {
// body of dmInitCallBack
});